pandas/core/dtypes/dtypes.py

```diff
@@ -420,6 +421,11 @@ def _hash_categories(categories, ordered=True):
                     # find a better solution
                     hashed = hash((tuple(categories), ordered))
                     return hashed
+
+            if is_datetime64tz_dtype(categories.dtype):
```

Comment:
can you ``categories.to_numpy()`` always?

Reply:
Hmm, possibly. We'll still need the special case for datetime64tz_dtype to pass `dtype=_NS_DTYPE`, since `Index[datetime64[ns, tz]].to_numpy()` returns an ndarray of Timestamp objects.

Reply:
maybe add a TODO here, this is kind of special casing